matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/LoadTimelineStrategy.kt

```diff
@@ -240,10 +249,50 @@ internal class LoadTimelineStrategy constructor(
 
     fun buildSnapshot(): List<TimelineEvent> {
         val events = buildSendingEvents() + timelineChunk?.builtItems(includesNext = true, includesPrev = true).orEmpty()
+        val eventsUpdated = updateRepliedEventIfNeeded(events)
```

Comment:
I'd avoid using map here, will slow down a lot timeline when there is a lot of events already loaded. It should be handled directly inside the buildEvents